src/libtest/lib.rs

```diff
@@ -1379,44 +1418,123 @@ pub fn run_test(
+        match strategy {
+            RunStrategy::InProcess(tx) | RunStrategy::SpawnPrimary(tx) => {
+                tx.send((desc, TrIgnored, Vec::new())).unwrap();
+            }
+            RunStrategy::SpawnedSecondary => panic!(),
+        }
         return;
     }
 
     fn run_test_inner(
         desc: TestDesc,
-        monitor_ch: Sender<MonitorMsg>,
         nocapture: bool,
+        strategy: RunStrategy,
         testfn: Box<dyn FnOnce() + Send>,
         concurrency: Concurrent,
     ) {
         // Buffer for capturing standard I/O
         let data = Arc::new(Mutex::new(Vec::new()));
-        let data2 = data.clone();
 
+        // TODO: Add RunStrategy. Maybe get rid of CaptureMode or make it
+        // internal to this function.
+        //
+        // RunStrategy::InProcess => callback will include output
+        // RunStrategy::Spawn => same from perspective of caller
+        // RunStrategy::SpawnedSecondary => output to stderr; will exit after
+        //                                  reporting result
         let name = desc.name.clone();
         let runtest = move || {
-            let oldio = if !nocapture {
-                Some((
-                    io::set_print(Some(Box::new(Sink(data2.clone())))),
-                    io::set_panic(Some(Box::new(Sink(data2)))),
-                ))
-            } else {
-                None
-            };
+            match strategy {
+                RunStrategy::InProcess(monitor_ch) => {
+                    let oldio = if !nocapture {
+                        Some((
+                            io::set_print(Some(Box::new(Sink(data.clone())))),
+                            io::set_panic(Some(Box::new(Sink(data.clone())))),
+                        ))
+                    } else {
+                        None
+                    };
+
+                    let result = catch_unwind(AssertUnwindSafe(testfn));
+
+                    if let Some((printio, panicio)) = oldio {
+                        io::set_print(printio);
+                        io::set_panic(panicio);
+                    }
 
-            let result = catch_unwind(AssertUnwindSafe(testfn));
+                    let test_result = match result {
+                        Ok(()) => calc_result(&desc, Ok(())),
+                        Err(e) => calc_result(&desc, Err(e.as_ref())),
+                    };
+                    let stdout = data.lock().unwrap().to_vec();
+                    monitor_ch.send((desc.clone(), test_result, stdout)).unwrap();
+                }
 
-            if let Some((printio, panicio)) = oldio {
-                io::set_print(printio);
-                io::set_panic(panicio);
-            };
+                RunStrategy::SpawnPrimary(monitor_ch) => {
+                    let args = env::args().collect::<Vec<_>>();
+                    let output = match Command::new(&args[0])
+                        .args(&args[1..])
```

Comment:
I think I'm forgetting something but what other arguments need to be passed?

Reply:
None! Removed this.